### ISSUE TITLE
core-js seems to sometimes cause a "Maximum call stack size exceeded"

### ISSUE THREAD

**Issue by stefanpenner** [6to5/6to5#394]
I haven't be able to produce a minimal reproduction yet, but intend to later this evening.

It appears, when requiring 6to5 is deferred to later during the lifecycle of ember-cli (to mitigate the 250ms load cost) core-js becomes unhappy

stack trace

```
Maximum call stack size exceeded
RangeError: Maximum call stack size exceeded
    at defineProperty.set (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:543:22)
    at simpleSet (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:367:15)
    at defineProperty.set (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:544:11)
    at simpleSet (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:367:15)
    at defineProperty.set (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:544:11)
    at simpleSet (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:367:15)
    at defineProperty.set (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:544:11)
    at simpleSet (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:367:15)
    at defineProperty.set (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:544:11)
    at simpleSet (/Users/stefan/src/broccoli-6to5-transpiler/node_modules/6to5/node_modules/core-js/library.js:367:15)
```

**Comment by sebmck** [comment]
@stefanpenner Input code?

**Comment by stefanpenner** [comment]
@sebmck it seems to be unique to requiring `6to5` on-demand, rather then eagerly. Some-what perplexing.... 

Basic attempts to reproduce have thus far failed. I need to grab some dinner, i'll attempt to isolate after.

**Comment by thejameskyle** [comment]
ping @zloirock

**Comment by zloirock** [comment]
I don't use broccoli & ember, show me broken code.

**Comment by sebmck** [comment]
It's likely that before `core-js` is required, some other module has gone and messed with prototypes or loaded their own broken polyfill.

**Comment by zloirock** [comment]
Such may be due to a broken `Obect.defineProperty`.

**Comment by sebmck** [comment]
@stefanpenner I've got a blank ember app setup. What are the steps/commands I have to run now to reproduce?

**Comment by stefanpenner** [comment]
starting place:

**Comment by stefanpenner** [comment]
starting point:

```
git clone https://github.com/stefanpenner/___6to5
```

and run `ember build` this will work correctly.

If you then upgrade:  `node_modules/ember-cli-6to5/node_modules/broccoli-6to5-transpiler/package.json` 6to5 to `^2.5.0` from `^1.7.14` run the appropriate `npm i` and try `ember build` again, the above error should occur.

**Comment by stefanpenner** [comment]
interestingly, if https://github.com/gordonkristan/ember-cli-6to5/blob/master/index.js#L12 is eagerly required (at the top of the file) no such problem arises. Unfortunately the 250ms - 300ms boot penalty of requiring 6to5 eagerly has lead to the late requiring

**Comment by sebmck** [comment]
`simpleSet` is being recursively set with `(Symbol.prototype, Symbol.toStringTag, "Symbol)`.

@zloirock Any ideas on where I should be looking?

**Comment by zloirock** [comment]
@sebmck I think, it's broken `Object.defineProperty`. Try `core-js/client/library` instead of `core-js/library`.

**Comment by sebmck** [comment]
Found it. You were right @zloirock.

It's in `___6to5/node_modules/ember-cli/node_modules/es6-module-transpiler/dist/es6-module-transpiler.js`.

```javascript
function defineProperty(object, name, descriptor) {
  if (NameModule.isName(name)) {
    // Private names should never be enumerable.
    if (descriptor.enumerable) {
      descriptor = Object.create(descriptor, {
        enumerable: {value: false}
      });
    }
    $defineProperty(object, name[internalStringValueName], descriptor);
  } else {
    assertNotName(name);
    $defineProperty(object, name, descriptor);
  }
}
```

**Comment by zloirock** [comment]
Current Traceur wrapper for `defineProperty` in REPL differs from it and passes this test. I added fallback for this case, it will be available in next version `core-js`.

**Comment by zloirock** [comment]
https://github.com/google/traceur-compiler/issues/398

**Comment by sebmck** [comment]
Fixed by 2a1e0122403458446440d4f806b1c5b5ced68f2e and released as of 2.7.1

**Comment by stefanpenner** [comment]
Awesome thanks for the quick turn around
